### PR TITLE
mobile: Remove TestEngineBuilder

### PR DESCRIPTION
**Pull request by fredyw**
All tests have been updated to not use `TestEngineBuilder`, so it can be safely removed.

Risk Level: low (code is no longer used)
Testing: unit tests
Docs Changes: n/a
Release Notes: n/a
Platform Specific Features: n/a

### CODE REVIEW

File: mobile/test/common/integration/base_client_integration_test.cc [hunk 1]
```diff
@@ -203,16 +203,6 @@ void BaseClientIntegrationTest::createEnvoy() {
     }
   }
 
-  if (override_builder_config_) {
```
Comment (abeyad) [inline mobile/test/common/integration/base_client_integration_test.cc:206]: I don't think `override_builder_config_` is used anywhere anymore now after this PR, so we can remove it from the header file (https://github.com/envoyproxy/envoy/blob/6f95c27ef7a65f74a0a44783068690c7bf10f993/mobile/test/common/integration/base_client_integration_test.h#L80)

**Review by abeyad — APPROVED** [review]
Woohoo!

**Review by RyanTheOptimist — COMMENTED** [review]
LOVE THIS!

File: mobile/test/common/integration/base_client_integration_test.cc (hunk 1 above)
Comment (fredyw) [reply mobile/test/common/integration/base_client_integration_test.cc:206]: This is already removed.

**Review by abeyad — APPROVED** [review]

**Pull request merged**